src/digraph.rs

```diff
@@ -1150,6 +1152,120 @@ impl PyDiGraph {
         }
     }
 
+    /// Read an edge list file and create a new PyDiGraph object from the
+    /// contents
+    ///
+    /// The expected format for the edge list file is a line seperated list
+    /// of deliminated node ids. If there are more than 3 elements on
+    /// a line the 3rd on will be treated as a string weight for the edge
+    ///
+    /// :param str path: The path of the file to open
+    /// :param str comment: Optional character to use as a comment by default
+    ///     there are no comment character
+    /// :param str deliminator: Optional character to use as a deliminator by
+    ///     default any whitespace will be used
+    ///
+    /// For example:
+    ///
+    /// .. jupyter-execute::
+    ///
+    ///   import os
+    ///   import tempfile
+    ///
+    ///   from PIL import Image
+    ///   import pydot
+    ///
+    ///   import retworkx
+    ///
+    ///
+    ///   with tempfile.NamedTemporaryFile('wt') as fd:
+    ///       path = fd.name
+    ///       fd.write('0 1\n')
+    ///       fd.write('0 2\n')
+    ///       fd.write('0 3\n')
+    ///       fd.write('1 2\n')
+    ///       fd.write('2 3\n')
+    ///       fd.flush()
+    ///       graph = retworkx.PyDiGraph.read_edge_list(path)
+    ///
+    ///   # Draw graph
+    ///   dot = pydot.graph_from_dot_data(graph.to_dot())[0]
+    ///
+    ///   with tempfile.TemporaryDirectory() as tmpdirname:
+    ///       tmp_path = os.path.join(tmpdirname, 'dag.png')
+    ///       dot.write_png(tmp_path)
+    ///       image = Image.open(tmp_path)
+    ///       os.remove(tmp_path)
+    ///   image
+    ///
+    #[staticmethod]
+    #[text_signature = "(path, /, comment=None, deliminator=None)"]
+    pub fn read_edge_list(
```

Comment:
That's a very good question, I didn't do it explicitly and completely forgot about it. I think it will error on a blank lines (when trying to access the target from the split string) . Let me add a test to confirm that and fix it.

Reply:
Ok, fixed this in: https://github.com/Qiskit/retworkx/pull/113/commits/d9bb203d6555039a62a6c602b0d321a40c77d4d4